martin/src/cog/source.rs

```diff
@@ -315,95 +260,20 @@ fn verify_requirements(
-        .with_limits(tiff::decoder::Limits::unlimited());
-    let model = ModelInfo::decode(&mut decoder, path);
-    let origin = get_origin(
-        model.tie_points.as_deref(),
-        model.transformation.as_deref(),
-        path,
-    )?;
-    let (full_width_pixel, full_length_pixel) = decoder.dimensions().map_err(|e| {
-        CogError::TagsNotFound(
-            e,
-            vec![Tag::ImageWidth.to_u16(), Tag::ImageLength.to_u16()],
-            0, // we are at ifd 0, the first image, haven't seek to others
-            path.clone(),
-        )
-    })?;
-    let full_resolution = get_full_resolution(
-        model.pixel_scale.as_deref(),
-        model.transformation.as_deref(),
-        path,
-    )?;
-    let full_width = full_resolution[0] * f64::from(full_width_pixel);
-    let full_length = full_resolution[1] * f64::from(full_length_pixel);
-    let extent = get_extent(
-        &origin,
-        model.transformation.as_deref(),
-        (full_width_pixel, full_length_pixel),
-        (full_width, full_length),
-    );
-    verify_requirements(&mut decoder, &model, path)?;
-    let mut zoom_and_ifd: HashMap<u8, usize> = HashMap::new();
-    let mut zoom_and_tile_across_down: HashMap<u8, (u32, u32)> = HashMap::new();
-
-    let nodata: Option<f64> = if let Ok(no_data) = decoder.get_tag_ascii_string(GdalNodata) {
-        no_data.parse().ok()
-    } else {
-        None
-    };
-
-    let images_ifd = get_images_ifd(&mut decoder, path);
-
-    for (idx, image_ifd) in images_ifd.iter().enumerate() {
-        decoder
-            .seek_to_image(*image_ifd)
-            .map_err(|e| CogError::IfdSeekFailed(e, *image_ifd, path.clone()))?;
-
-        let zoom = u8::try_from(images_ifd.len() - (idx + 1))
-            .map_err(|_| CogError::TooManyImages(path.clone()))?;
-
-        let (tiles_across, tiles_down) = get_grid_dims(&mut decoder, path, *image_ifd)?;
-
-        zoom_and_ifd.insert(zoom, *image_ifd);
-        zoom_and_tile_across_down.insert(zoom, (tiles_across, tiles_down));
-    }
-
-    if images_ifd.is_empty() {
-        Err(CogError::NoImagesFound(path.clone()))?;
-    }
-
-    Ok(Meta {
-        min_zoom: 0,
-        max_zoom: images_ifd.len() as u8 - 1,
-        model,
-        origin,
-        extent,
-        zoom_and_ifd,
-        zoom_and_tile_across_down,
-        nodata,
-    })
-}
-
 fn get_grid_dims(
     decoder: &mut Decoder<File>,
     path: &Path,
     image_ifd: usize,
 ) -> Result<(u32, u32), FileError> {
     let (tile_width, tile_height) = (decoder.chunk_dimensions().0, decoder.chunk_dimensions().1);
-    let (image_width, image_length) = get_image_dims(decoder, path, image_ifd)?;
+    let (image_width, image_length) = dim_in_pixel(decoder, path, image_ifd)?;
     let tiles_across = image_width.div_ceil(tile_width);
     let tiles_down = image_length.div_ceil(tile_height);
 
     Ok((tiles_across, tiles_down))
 }
 
-fn get_image_dims(
+fn dim_in_pixel(
```

Comment:
I think that hamming coding is a lot harder to read 
Personal preference mostly, feel free to ignore 

```suggestion
/// Decodes the files' `(width, height)` in pixels
fn decode_image_dimensions_in_pixels(
```

Reply:
`decode_image_dimensions_in_pixels` is good but a little bit too long I guess.. I'm still thinking about it.

Reply:
Updated to `dimensions_in_pixel` and `deminesions_in_model` now. And more docs added. Yeah not good enough I know. Naming things is hard....🫠